measureme/src/stringtable.rs

```diff
@@ -187,11 +180,10 @@ impl<'s> StringComponent<'s> {
                 &mut bytes[s.len()..]
             }
             StringComponent::Ref(string_id) => {
-                assert!(string_id.0 == string_id.0 & STRING_ID_MASK);
-                let tagged = string_id.0 | (1u32 << 31);
-
-                &mut bytes[0..4].copy_from_slice(&tagged.to_be_bytes());
-                &mut bytes[4..]
+                assert!(STRING_REF_ENCODED_SIZE == 5);
```

Comment:
Is this the assert we actually want here? (It's trivially true and rustc actually seems to optimize it out before codegen even occurs but I'm not sure why we'd assert that in this method.)

Reply:
Yeah, the `assert!` is supposed to be a self-testing explanation of the magic numbers below. I think the best thing to do is (1) leaving the assertion in, (2) adding a comment that explains its purpose, and (3) applying the changes you suggest below.

For this kind of low-level byte shuffling stuff I like to add these kinds of tripwires that decrease the possibility of overlooking necessary adaptations when the encoding changes.

Reply:
I added a comment but ended up *not* replacing the literal `5` with `STRING_REF_ENCODED_SIZE`. I don't know about you but writing `STRING_REF_ENCODED_SIZE` did not actually seem to make the code more readable.

What might make it more readable would be to have a bunch of constants like `STRING_REF_TAG_OFFSET`, `STRING_REF_VALUE_START_OFFSET`, and `STRING_REF_VALUE_END_OFFSET`, so that the encoding is kind of explained in one place. I don't think that's quite worth the trouble though.

Reply:
That seems reasonable to me 👍